tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs

```diff
@@ -1301,5 +1302,109 @@ public async Task MixedUserAndSystemAssignedManagedIdentityTestAsync()
                 Assert.AreEqual(SystemAssignedClientId, systemAssignedTokens[0].ClientId, "System-assigned ClientId mismatch in cache.");
             }
         }
+
+        [DataTestMethod]
+        [DataRow(ManagedIdentitySource.AppService, TestConstants.AppServiceEndpoint)]
+        [DataRow(ManagedIdentitySource.AzureArc, TestConstants.AzureArcEndpoint)]
+        [DataRow(ManagedIdentitySource.CloudShell, TestConstants.CloudShellEndpoint)]
+        [DataRow(ManagedIdentitySource.Imds, TestConstants.ImdsEndpoint)]
+        [DataRow(ManagedIdentitySource.MachineLearning, TestConstants.MachineLearningEndpoint)]
+        [DataRow(ManagedIdentitySource.ServiceFabric, TestConstants.ServiceFabricEndpoint)]
+        public async Task SAMIFails500PermanentlyAndRetryPolicyLifeTimeIsPerRequestAsync(ManagedIdentitySource managedIdentitySource, string endpoint)
+        {
+            using (new EnvVariableContext())
+            using (var httpManager = new MockHttpManager(isManagedIdentity: true))
+            {
+                var miBuilder = ManagedIdentityApplicationBuilder.Create(ManagedIdentityId.SystemAssigned)
+                    .WithHttpManager(httpManager);
+
+                // Disable cache to avoid pollution
+                miBuilder.Config.AccessorOptions = null;
+
+                var mi = miBuilder.Build();
+
+                // Simulate permanent 500s (to trigger the maximum number of retries)
+                int NUM_500 = ManagedIdentityRequest.DEFAULT_MANAGED_IDENTITY_MAX_RETRIES + 1; // initial request + maximum number of retries (3)
```

Comment:
Nit: rename to num500 since this is a local variable.

Reply:
I've since refactored this as a `const`, therefore I believe I'm supposed to make this PascalCase.